src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/SuccessLargeAndOrderedListProductIntegration.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Pim\Bundle\ApiBundle\tests\integration\Controller\Product;
+
+use Doctrine\Common\Collections\Collection;
+use Pim\Component\Catalog\Model\ProductInterface;
+use Pim\Component\Catalog\tests\integration\Normalizer\NormalizedProductCleaner;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * We want to test the API is capable of returning an ordered list of 100 items.
+ * ie, twice the size of a cursor page
+ */
+class SuccessLargeAndOrderedListProductIntegration extends AbstractProductTestCase
+{
+    /** @var Collection */
+    private $products;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        if (1 === self::$count) {
+
+            $identifiers = [];
+            for ($i = 0; $i < $this->getListSize(); $i++) {
+                $identifiers[] = 'sku-' . str_pad($i, 4, '0', STR_PAD_LEFT);
+            }
+
+            foreach ($identifiers as $identifier) {
+                $product = $this->createProduct($identifier, []);
+                $this->products[$product->getId()] = $product;
+            }
+            // the API will return products sorted by MySQL ID, and that's what we expect
+            // for instance, if we have 100 products
+            // 1, 10, 100, 11, 12, 13, 14, 15, 16, 17, 18, 19, 2, 20, 21...
+            ksort($this->products, SORT_STRING);
+        }
+    }
+
+    public function testPaginationAllProducts()
+    {
+        $standardizedProducts = [];
+        foreach ($this->products as $product) {
+            $standardizedProducts[] = $this->getStandardizedProduct($product->getIdentifier());
+        }
+        $standardizedProducts = implode(',', $standardizedProducts);
```

Comment:
I prefer `json_encode`and then later, `"item": $standardizedProducts`, but maybe  a matter of taste.